kernel.condvar build failure for intel_adsp

### Describe the bug

`kernel.condvar` (`tests/kernel/condvar/condvar_api`) fails for `intel_adsp/ace40/nvl/sim` as a build failure.

This failure is observed with `xt-clang` toolchain and zephyr toolchain (on ptl).  

The first bad commit identified by bisect is:
5d8eec009ed02e6d5e8dcac0a2690c9fd90ed8de

After this commit, the testcase no longer builds on this platform and fails during
userspace object metadata generation:

- `Too many thread objects (19)`
- `Increase CONFIG_MAX_THREAD_BYTES to 3`

### Regression

- [x] This is a regression.

### Steps to reproduce

```sh
scripts/twister -p intel_adsp/ace40/nvl/sim \
  -T tests/kernel/condvar/condvar_api \
  --test kernel.condvar \
  --inline-logs -v \
  --retry-failed 1 --retry-interval 15 --retry-build-errors
```

### Relevant log output

```shell
[1110/1137] Generating app_smem_unaligned linker section
[1111/1137] Generating linker_zephyr_pre0.cmd
[1112/1137] Building C object zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj
[1113/1137] Linking C executable zephyr/zephyr_pre0.elf
[1114/1137] Generating app_smem_aligned linker section
[1115/1137] Generating kobject_prebuilt_hash.gperf
FAILED: zephyr/kobject_prebuilt_hash.gperf intel_adsp_ace40_nvl_sim/xt-clang/tests/kernel/condvar/condvar_api/kernel.condvar/zephyr/kobject_prebuilt_hash.gperf
cd intel_adsp_ace40_nvl_sim/xt-clang/tests/kernel/condvar/condvar_api/kernel.condvar/zephyr && python3 scripts/build/gen_kobject_list.py --kernel intel_adsp_ace40_nvl_sim/xt-clang/tests/kernel/condvar/condvar_api/kernel.condvar/zephyr/zephyr_pre0.elf --gperf-output kobject_prebuilt_hash.gperf --include-subsystem-list intel_adsp_ace40_nvl_sim/xt-clang/tests/kernel/condvar/condvar_api/kernel.condvar/zephyr/misc/generated/struct_tags.json
Too many thread objects (19)
Increase CONFIG_MAX_THREAD_BYTES to 3
ninja: build stopped: subcommand failed.
```

### Impact

Major – Severely degrades functionality; workaround is difficult or unavailable.

### Environment

- OS: Linux
- Toolchain: xt-clang
- Platform: `intel_adsp/ace40/nvl/sim`
- Test: `kernel.condvar` (`tests/kernel/condvar/condvar_api`)
- First failing build version observed: `v4.4.0-6234-g5d8eec009ed0`

### Additional Context

_No response_

--- a/tests/kernel/condvar/condvar_api/Kconfig
+++ b/tests/kernel/condvar/condvar_api/Kconfig
@@ -1,0 +1,8 @@
+# Copyright (c) The Zephyr Project Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+config MAX_THREAD_BYTES
+	default 3 if USERSPACE && SMP
+
+# Include Zephyr's Kconfig.
+source "Kconfig.zephyr"

